Mathlib/NumberTheory/FLT/Three.lean

```diff
@@ -0,0 +1,40 @@
+/-
+Copyright (c) 2024 Riccardo Brasca. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Riccardo Brasca
+-/
+import Mathlib.Data.ZMod.Basic
+
+/-!
+# Fermat Last Theorem in the case `n = 3`
+The goal of this file is to prove Fermat Last theorem in the case `n = 3`.
+
+## Main results
+* `fermatLastTheoremThree_case_1`: the first case of Fermat Last Theorem when `n = 3`:
+  if `a b c : ℕ` are such that `¬ 3 ∣ a * b * c`, then `a ^ 3 + b ^ 3 ≠ c ^ 3`.
+
+## TODO
+Prove case 2.
+-/
+
+open ZMod
+
+private lemma cube_of_castHom_ne_zero {n : ZMod 9} :
+    castHom (show 3 ∣ 9 by norm_num) (ZMod 3) n ≠ 0 → n ^ 3 = 1 ∨ n ^ 3 = 8 := by
+  fin_cases n <;> decide
+
+private lemma cube_of_not_dvd {n : ℕ} (h : ¬ 3 ∣ n) :
+    (n : ZMod 9) ^ 3 = 1 ∨ (n : ZMod 9) ^ 3 = 8 := by
+  apply cube_of_castHom_ne_zero
+  rwa [map_natCast, Ne, Fin.nat_cast_eq_zero]
+
+/--If `a b c : ℕ` are such that `¬ 3 ∣ a * b * c`, then `a ^ 3 + b ^ 3 ≠ c ^ 3`. -/
+theorem fermatLastTheoremThree_case_1 {a b c : ℕ} (hdvd : ¬ 3 ∣ a * b * c) :
+    a ^ 3 + b ^ 3 ≠ c ^ 3 := by
```

Comment:
Can you please restate this using `FermatLastTheoremFor 3` using `Mathlib/NumberTheory/FLT/Basic.lean`?

Reply:
Not really, this is only a particular case: `FermatLastTheoremFor 3` is the statement
```lean
  ∀ a b c : α, a ≠ 0 → b ≠ 0 → c ≠ 0 → a ^ n + b ^ n ≠ c ^ n
```
but I cannot write `FermatLastTheoremFor 3 a b c` to specialize to the variable I declared before (and I have to do it since I need to assume `¬ 3 ∣ a * b * c`. Of course at the end I plan to use `FermatLastTheoremFor 3` for the final statement.

Reply:
Aah, nifty dependent types (-;